akka-actor/src/main/scala/akka/actor/Actor.scala

```diff
@@ -363,7 +363,7 @@ object Actor {
    * Default placeholder (null) used for "!" to indicate that there is no sender of the message,
    * that will be translated to the receiving system's deadLetters.
    */
-  final val noSender = null.asInstanceOf[ActorRef] // avoid type ascription: compile-time constant
+  final val noSender = null: ActorRef // avoid type ascription: compile-time constant
```

Comment:
Hmm, you’re right, I assumed too much: this is not actually a constant. It only is a constant if it has type `Null`, which incidentally also does work.